Compilation fails on ARM and gcc 4.9

```shell
   CC     cache.o
In file included from cache.h:23:0,
                 from cache.c:19:
locking.h:37:41: error: initializer element is not constant
 #define FIFO_MUTEX_INITIALIZER ((struct fifo_mutex) { \
                                         ^
cache.c:86:33: note: in expansion of macro 'FIFO_MUTEX_INITIALIZER'
 struct fifo_mutex cache_mutex = FIFO_MUTEX_INITIALIZER;
                                 ^
````

Switching to `GCC 5.4` seems to avoid the error.

--- a/locking.h
+++ b/locking.h
@@ -34,10 +34,10 @@ extern pthread_t main_thread;
 #define CMUS_COND_INITIALIZER PTHREAD_COND_INITIALIZER
 #define CMUS_RWLOCK_INITIALIZER PTHREAD_RWLOCK_INITIALIZER
 
-#define FIFO_MUTEX_INITIALIZER ((struct fifo_mutex) { \
+#define FIFO_MUTEX_INITIALIZER { \
 		.mutex = PTHREAD_MUTEX_INITIALIZER, \
 		.tail = ATOMIC_VAR_INIT(NULL), \
-	})
+	}
 
 void cmus_mutex_lock(pthread_mutex_t *mutex);
 void cmus_mutex_unlock(pthread_mutex_t *mutex);

